merkle/log_verifier_test.go

```diff
@@ -411,7 +433,27 @@ func TestVerifyConsistencyProof(t *testing.T) {
 			roots[snapshot1-1].h,
 			roots[snapshot2-1].h, proof)
 		if err != nil {
-			t.Fatalf("Failed to verify known good proof for i=%d: %s", i, err)
+			t.Errorf("Failed to verify known good proof for i=%d: %s", i, err)
+		}
+	}
+}
+
+func TestVerifyConsistencyProofGenerated(t *testing.T) {
+	size := int64(130)
+	tree, v := createTree(size)
+	roots := make([][]byte, size+1)
+	for i := int64(0); i <= size; i++ {
+		roots[i] = tree.RootAtSnapshot(i).Hash()
+	}
+
+	for i := int64(0); i <= size; i++ {
+		for j := i; j <= size; j++ {
+			proof := rawProof(tree.SnapshotConsistency(i, j))
+			t.Run(fmt.Sprintf("size:%d:consistency:%d-%d", size, i, j), func(t *testing.T) {
```

Comment:
Should we worry about 8515 sub-tests here? If something breaks, this will probably generate quite a big log.

Reply:
It might be an issue, it's many more than we usually have. I don't know TBH.